hilti/runtime/src/types/time.cc

```diff
@@ -18,6 +18,24 @@ Time time::current_time() {
     return Time(t, Time::SecondTag());
 }
 
+Time time::mktime(uint64_t y, uint64_t m, uint64_t d, uint64_t H, uint64_t M, uint64_t S) {
+    struct tm t;
+    t.tm_sec = S;
+    t.tm_min = M;
+    t.tm_hour = H;
+    t.tm_mday = d;
+    t.tm_mon = m - 1;
+    t.tm_year = y - 1900;
+    t.tm_isdst = -1;
```

Comment:
Should we also allow users to override DST and `tm_zone`? Without giving them this option this function seem to be hard to use to convert general time stamps which differ from the setting of the particular machine this was running on.

Same question for the function exposed in the Spicy API.

Reply:
mktime(3) always seems to use local time, with `dst` and `tm_zone` being non-portable extensions if considered at all. We could use `timegm()` instead, which takes UTC, but that would lead to unexpected values too I think. I don't see an easy alternative, guess we could consider finding a different implementation of `mktime` that provides a TZ parameter.

Reply:
I still stand by my objection that this is hard to use in general, but I didn't mean to suggest that we need to provide a full set of time functions. Looking at [Python](https://docs.python.org/3/library/time.html) they seem to (at least at that level) mirror POSIX; [Golang](https://golang.org/pkg/time/) reinvents this, but I am unsure what shape their wheel has.

As an idea, we could accept optional DST and TZ parameters here and perform the conversion inside the function if they are given, or we follow whatever Zeek choose to follow (since they currently are our biggest user and maybe already needed to solve for this).

Reply:
Yeah, I think we'd need to do what you suggest and at least accept a `TZ` parameter. But I'm not sure that doing that conversion is straight-forward. If you have an idea how to implement that easily, happy to add; otherwise I'd go with this version for now and we can add a defaulted `TZ` parameter later when need comes up (`dst` I'd just skip).

Reply:
Going ahead with this, as discussed.